src/envoy/http/authn/http_filter.cc

```diff
@@ -65,7 +67,9 @@ FilterHeadersStatus AuthenticationFilter::decodeHeaders(HeaderMap& headers,
       filter_config_.policy().origin_is_optional();
 
   if (!success) {
-    rejectRequest("Origin authentication failed.");
+    rejectRequest(
```

Comment:
I would like to propagate jwt auth error reason to the response, not just about wording.

Reply:
An authn policy may contain multiple origin authentication policies and/or multiple peer authentication policies. The rejection message does not have a mTLS/JWT failure code since the authentication fails when none of the mTLS/JWT policies are satisfied (not due to a single mTLS/JWT policy failure).

Reply:
We need to propagate it, and pack the message in a way that make sense (can even be focus on the single JWT, which is the major use cases) 
The reason for this issue is it is very hard to debug why jwt doesn't work.

Reply:
The authn policy is deployed on the service side whereas the rejection message is sent to a client. Sending a rejection message with detailed and combined JWT diagnosis information to a client may not necessarily help clients because clients are mere users of the service, have no control of the service policies, and are generally non-technical. For example, a smart phone may call tens of services but the smartphone owner likely never read the detailed authn rejection diagnosis message unless he/she is a developer of the service, who can use logs to debug the authn failures and get detailed diagnosis information. Even worse, since such detailed authn rejection message is sent to any client, an attacker may potentially take advantage of such detailed authn failure rejection messages to debug the attacks.